src/coreclr/tools/Common/TypeSystem/Common/ExceptionStringID.cs

```diff
@@ -41,5 +41,9 @@ public enum ExceptionStringID
         // BadImageFormatException
         BadImageFormatGeneric,
         BadImageFormatSpecific,
+
+        // MarshalDirectiveException
+        MarshalDirectiveGeneric,
+        MarshalDirectiveInvalidSizeParamIndex,
```

Comment:
I am not sure whether we need to special case `MarshalDirectiveInvalidSizeParamIndex` here. I think using `MarshalDirectiveGeneric` for this case would be good enough.

Reply:
I personally think that Marshal is complicated enough to have one blanket exception. If at least some areas produce slightly different exceptions it helpful for troubleshooting in my experience. But if you think that's add too much complexity, I will remove that.

Reply:
We have the [ILVerify tool](https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/ILVerify) that can be used to validate inputs and provide a better experience around it. It could validate p/invoke signatures too. It can provide a lot more granular detailed output than this compiler. I would shoot for the minimally viable bar here.

(I was actually even considering just disabling the parts of the tests that test the invalid p/invokes. I will probably do that with some non-interop tests because we try to be compatible "within reason" - we don't go out of way to be 100% compatible.)

Reply:
Thanks for explanation.